### PR TITLE
testcases: template: kvm-unit-tests: lockdown to a GIT_REF

### PR DESCRIPTION
**Pull request by roxell**
Signed-off-by: Anders Roxell <anders.roxell@linaro.org>

### CODE REVIEW

**Comment by nareshkamboju** [comment]
I would like to bump the git commit-id in my next patch.

**Comment by roxell** [comment]
that make sense but show we do that in a followup patch?

**Comment by nareshkamboju** [comment]
@roxell 
Here is the GIT_REF id
f2606a873e805f9aff4c4879ec75e65d7e30af73

Ref:
https://git.kernel.org/pub/scm/virt/kvm/kvm-unit-tests.git/commit/?id=f2606a873e805f9aff4c4879ec75e65d7e30af73

**Comment by mwasilew** [comment]
Wouldn't it be better to pass GIT_REF as a parameter? This way you won't have to change it in the template every time you need to update the version.

**Comment by roxell** [comment]
new patch done. That makes a lot of sense to do @mwasilew thank you for the idea.

File: testcases/master/template-kvm-unit-tests.yaml.jinja2 [hunk 1]
```diff
@@ -24,4 +24,5 @@
       parameters:
         SKIP_INSTALL: 'true'
         SMP: {% if DEVICE_TYPE == "juno-r2" %}'false'{% else %}'true'{% endif %}
+        GIT_REF: "{{KVM_UNIT_TESTS_REVISION | default('perkele_need_to_specify_something')}}"
```
Comment (mwasilew) [inline testcases/master/template-kvm-unit-tests.yaml.jinja2]: While I like the default a lot, maybe it should just be 'master' ?

Comment (roxell) [reply testcases/master/template-kvm-unit-tests.yaml.jinja2]: Maybe we should use this as a default for all the GIT_REF's and/or branches we try to checkout so the user have to do something active in order to get the sha/branch he/she wants?

Comment (mrchapp) [reply testcases/master/template-kvm-unit-tests.yaml.jinja2]: We want defaults to work, not to fail.

Comment (nareshkamboju) [reply testcases/master/template-kvm-unit-tests.yaml.jinja2]: @roxell do not force the user for it.

Comment (nareshkamboju) [reply testcases/master/template-kvm-unit-tests.yaml.jinja2]: @roxell default as master is not harmful.

Comment (mwasilew) [reply testcases/master/template-kvm-unit-tests.yaml.jinja2]: I get it, when I want more comments on my patches I should start using swear words :)

Comment (roxell) [reply testcases/master/template-kvm-unit-tests.yaml.jinja2]: yes =)

**Comment by nareshkamboju** [comment]
This change looks good to me.

**Pull request merged**